articles/per_package_documentation.md

```diff
@@ -0,0 +1,115 @@
+---
+layout: default
+title: Per-Package Documentation
+permalink: articles/per_package_documentation.html
+abstract: This article describes the requirements and design of ROS 2’s per-package documentation system.
+author: Marya Belanger
+published: true
+---
+
+{:toc}
+
+# {{ page.title }}
+
+<div class="abstract" markdown="1">
+{{ page.abstract }}
+</div>
+
+Original Author: {{ page.author }}
+
+## Background
+
+ROS 2 is lacking a process for the conglomeration of package documentation.
+Discoverability for package documentation (and documentation in general) is one of the most frequent user complaints.
+
+The goal of this design is to standardize a documentation process for ROS 2 packages to encourage maintainers to robustly document their packages.
+The results will be presented in a consistent way alongside the general ROS 2 documentation, making the documentation as a whole discoverable and consistent.
+
+In general, our vision for the system is:
+
+- Package maintainers document their packages in their repositories following some guidelines or templates we recommend
+- Package documentation from the repositories is built on the ROS 2 infrastructure and deployed to the ROS 2 documentation site in an automatic process with no input from the maintainer besides an opt-in `doc` block in the distribution.yaml
+  - See [REP 141](https://www.ros.org/reps/rep-0141.html) for more context.
+- Package documentation is indexed alongside ROS 2's generic content on our docs site
+
+## Context
+
+The per-package documentation plan is an extension of the general ROS 2 documentation project.
+That project, in the implementation phase now, includes some keys points that affect the context of the per-package documentation.
+
+- All documentation will be hosted on `docs.ros.org`
+- All documentation will be versioned by ROS 2 distribution names
+- The URL structure will be `docs.ros.org/<lang>/<distro>/<page>` for generic documentation and `docs.ros.org/<lang>/<distro>/p/<package_name>/<page>` for package documentation
+  - For more context on where documentation will fall under this URL structure, see [this diagram](https://docs.google.com/drawings/d/1KxzDrcSZzwGgudk-kEXGWnoRuAtc1ffl_KBeIu-V60Y/edit?usp=sharing).
+
+## Requirements
+
+The "Primary requirements" are those that must be in place for the system to be functional and achieve its purpose.
+The "Secondary requirements" are important features and functionality that are not necessary to roll out the first stage of implementation, but should be carried out as soon as possible following the initial roll out.
+
+### Primary requirements
```

Comment:
An additional primary requirement I'd like to propose is that the format is a common markdown language (rst, md, etc) that can be rendered by GitHub - that will help catch a bunch of issues before release.

Reply:
Thanks for pointing this out, it's one of those things that we internally assume by default so we didn't consider, which is exactly why we wanted outside eyes on this!

We're going to default to rst